magicctapipe/scripts/lst1_magic/IT_container_data_MC_bash_scripts/LSTnsb_MC.py

```diff
@@ -0,0 +1,118 @@
+"""
+Evaluates NSB level for a LST run
+"""
+import argparse
+import glob
+import logging
+import os
+
+import numpy as np
+import yaml
+from lstchain.image.modifier import calculate_noise_parameters
+
+__all__ = ["nsb"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def nsb(run_list, simtel, lst_config, run_number):
+    """
+    Here we compute the NSB value for a run based on a subset of subruns.
+
+    Parameters
+    ----------
+    run_list: list
+        List of subruns in the run
+    simtel: str
+        Simtel (MC) file to be used to evaluate the extra noise in dim pixels
+    lst_config: str
+        LST configuration file (cf. lstchain)
+    run number: int
+        LST run number
+
+    """
+    noise = []
+    denominator = 25
+    if len(run_list) == 0:
+        return
+    if len(run_list) < denominator:
+        mod = 1
+    else:
+        mod = int(len(run_list) / denominator)
+    for ii in range(0, len(run_list)):
+        if mod == 0:
+            break
+        if ii % mod == 0:
+            try:
+                a, _, _ = calculate_noise_parameters(simtel, run_list[ii], lst_config)
+                noise.append(a)
+            except IndexError:
+                mod = int(len(run_list) / (denominator + 1))
+                logger.info(
+                    f"WARNING: a subrun caused an error in the NSB level evaluation for run {run_number}. Check reports before using it"

```

Comment:
can't we write out also the subrun number?

Reply:
done